src/plans/create/utils/submitMigrationPlan.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  HookModel,
+  type IoK8sApiCoreV1Secret,
+  NetworkMapModel,
+  SecretModel,
+  StorageMapModel,
+  type V1beta1Hook,
+  type V1beta1NetworkMap,
+  type V1beta1StorageMap,
+} from '@kubev2v/types';
+
+import { MigrationHookFieldId } from '../steps/migration-hooks/constants';
+import type { CreatePlanFormData } from '../types';
+
+import { addOwnerRefs } from './addOwnerRefs';
+import { createDecryptionSecret } from './createDecryptionSecret';
+import { createMigrationHooks } from './createMigrationHooks';
+import { createNetworkMap } from './createNetworkMap';
+import { createPlan } from './createPlan';
+import { createStorageMap } from './createStorageMap';
+
+/**
+ * Handles the migration plan submission process including creation of network map,
+ * storage map, encryption secret, hooks, and establishing owner references.
+ */
+export const submitMigrationPlan = async (formData: CreatePlanFormData): Promise<void> => {
+  const {
+    diskDecryptionPassPhrases,
+    existingNetworkMap,
+    existingStorageMap,
+    migrationType,
+    networkMap: newNetworkMap,
+    networkMapName,
+    planName,
+    planProject,
+    postMigrationHook,
+    preMigrationHook,
+    preserveStaticIps,
+    rootDevice,
+    sharedDisks,
+    sourceProvider,
+    storageMap: newStorageMap,
+    storageMapName,
+    targetProject,
+    targetProvider,
+    transferNetwork,
+    vms,
+  } = formData;
+
+  let planNetworkMap = existingNetworkMap!;
+  let planStorageMap = existingStorageMap!;
+  let createdSecret: IoK8sApiCoreV1Secret | undefined = {};
+  let createdHooks: { preHook?: V1beta1Hook; postHook?: V1beta1Hook } = {};
+
+  // Collection of promises for concurrent resource creation
+  const createResourceRequests: Promise<void>[] = [];
+
+  // Create network map
+  if (!existingNetworkMap) {
+    createResourceRequests.push(
+      createNetworkMap({
+        mappings: newNetworkMap,
+        name: networkMapName,
+        planProject,
+        sourceProvider,
+        targetProvider,
+      }).then((networkMap: V1beta1NetworkMap) => {
+        planNetworkMap = networkMap;
```

Comment:
instead of using this method of then and saving to a let , as u already await for promise.all , u can extract the values as const [network, storage, secret] = await promise.all

Reply:
Good idea. Will update.